File: modules/wem-api/src/main/java/com/enonic/wem/api/query/aggregation/Aggregation.java
package com.enonic.wem.api.query.aggregation;

public abstract class Aggregation
{
    private final String name;

    public Aggregation( final Builder builder )
    {
        this.name = builder.name;
    }

    public String getName()
    {
        return name;
    }

    public static TermsAggregation.Builder terms()
    {
        return new TermsAggregation.Builder();
    }

    public static class Builder<T extends Builder>
    {
        private String name;

        public T name( final String name )
        {
            this.name = name;
            return (T) this;
        }

    }
}

File: modules/wem-webapp/src/main/webapp/admin/common/js/query/aggregation/Aggregation.ts
module api.query.aggregation {

    export class Aggregation {

        private name:string;

        constructor( builder:AggregationBuilder )
        {
            this.name = builder.name;
        }

        public getName():string
        {
            return this.name;
        }

        public static terms():TermsAggregationBuilder
        {
            return new TermsAggregationBuilder();
        }
    }

    export class AggregationBuilder {

        name:string;

        public setName( name:string ):AggregationBuilder
        {
            this.name = name;
            return this;
        }
    }
}
